app/models/agents/trust_pilot_agent.rb

```diff
@@ -0,0 +1,198 @@
+        {
+          "response_id": "59883aebfba87f08a8274e07",
+          "score": 5,
+          "comment": "Hello World",
+          "title": "Hello World …",
+          "language": "en",
+          "created_at": "2017-08-07T10:03:23Z",
+          "updated_at": "2017-08-07T10:03:23Z,
+          "company_reply": nil,
+          "is_verified": false,
+          "number_of_likes": 0,
+          "status": "active",
+          "report_data": nil,
+          "compliance_labels": [],
+          "consumer_name": "Jeffry Taraca",
+          "consumer_location": "Kirkeby, DK",
+          "email": "jhon@email.com",
+          "user_reference_id": "123"
+        }
+    MD
+
+    form_configurable :api_key
+    form_configurable :api_secret
+    form_configurable :business_units_ids
+    form_configurable :mode, type: :array, values: %w(all on_change merge)
+    form_configurable :access_token
+    form_configurable :refresh_token
+    form_configurable :expires_at
+    form_configurable :expected_update_period_in_days
+
+    def working?
+      event_created_within?(options['expected_update_period_in_days']) && !recent_error_logs?
+    end
+
+    def default_options
+      {
+        'api_key' => '{% credential TrustPilotApiKey %}',
+        'api_secret' => '{% credential TrustPilotApiSecret %}',
+        'mode' => 'on_change',
+        'expires_at' => Time.now.iso8601,
+        'expected_update_period_in_days' => '1'
+      }
+    end
+
+    def validate_options
+      %w[api_key api_secret access_token refresh_token expires_at business_units_ids mode expected_update_period_in_days].each do |key|
+        errors.add(:base, "The '#{key}' option is required.") if options[key].blank?
+      end
+
+      validate_web_request_options!
+    end
+
+    def check
+      prepare_request!
+      reviews = business_units.map(&:parse_reviews).flatten
+      reviews.each do |review|
+        if store_payload!(review)
+          create_event payload: review
+        end
+      end
+    end
+
+    private
+
+    def business_units
+      @business_units ||= business_units_ids.map do |bid|
+        business_unit = fetch_reviews(bid)
+        TrustPilotParser.new(business_unit)
+      end
+    end
+
+    def business_units_ids
+      @business_units_ids ||= interpolated['business_units_ids'].split(',').map(&:strip)
+    end
+
+    def fetch_reviews(business_unit_id)
+      url = "#{TRUSTPILOT_URL_BASE}/private/business-units/#{business_unit_id}/reviews"
+      fetch_resource(url)
+    end
+
+    def fetch_resource(uri)
+      response = faraday.run_request(:get, uri, nil, auth_headers)
+
+      unless response.success?
+        log "Fetch reviews Failed (#{response.status}) #{response.body}"
+        return {}
+      end
+
+      JSON.parse(response.body)
+    end
+
+    def store_payload!(review)
+      case interpolated['mode'].presence
+      when 'on_change'
+        review_id = review["id"]
+        if  old_events.find { |event| event.payload["id"] == review_id }
```

Comment:
maybe just `!old_events.find { |event| event.payload["id"] == review_id }.present?` ?)

Reply:
oh yeah is better, maybe with blank?